mkt/websites/management/commands/import_games_from_csv.py

```diff
@@ -116,15 +117,19 @@ def create_instances(self, data):
                     # Keywords use a M2M, so do that once the website is saved.
                     self.set_tags(website, row)
 
+                    WebsiteIndexer.index([website.id], no_delay=True)
+
                     # Launch task to fetch imgs once we know everything is OK.
                     try:
                         self.set_icon(website, row)
                         self.set_promo_imgs(website, row)
                     except Exception as e:
-                        website.delete()
                         print e
+                        website.delete()
```

Comment:
Maybe call `WebsiteIndexer.refresh_index()` before deleting to be sure if you still run into issues after changing the `index()` call as I suggest above